src/fe-gtk/sexy-spell-entry.c

```diff
@@ -1229,6 +1240,20 @@ sexy_spell_entry_changed(GtkEditable *editable, gpointer data)
 	sexy_spell_entry_recheck_all(entry);
 }
 
+static void
+sexy_spell_entry_preedit_changed(GtkEditable *editable, gchar* preedit, gpointer data)
+{
+	/** Because object internals are private in GTK 3 we need to store this when it changes. */
```

Comment:
I'd rather you just modify the objects struct.

Here is a port of this widget to GTK3: https://github.com/TingPing/libsexy3/blob/master/libsexy/sexy-spell-entry.c

See also: https://github.com/TingPing/libsexy3/blob/master/libsexy/sexy-spell-entry.c#L62